src/main/java/com/aws/greengrass/deployment/DeploymentService.java

```diff
@@ -363,13 +363,6 @@ private void finishCurrentDeployment() throws InterruptedException {
                     deploymentStatusKeeper.persistAndPublishDeploymentStatus(deploymentId, ggDeploymentId,
                             configurationArn, type, JobStatus.FAILED.toString(), statusDetails, rootPackages);
 
-                    if (currentDeploymentTaskMetadata.getDeploymentTask() instanceof KernelUpdateDeploymentTask) {
```

Comment:
hold on, why is this removed now? We didn't want if for REJECTED, but we should keep it here, I'm sure it is doing something useful

Reply:
I think something got mixed up here! will look at it again.